src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -2824,6 +2824,15 @@ void executeCodeBlockActions(ImmutableArray<SyntaxNode> executableCodeBlocks, IE
             SemanticModel semanticModel,
             AnalyzerExecutor analyzerExecutor)
         {
+            // Skip syntax nodes when analyzing record declaration constructor to avoid duplicate syntax node callbacks.
+            // We will analyze these nodes while analyzing the record declaration type symbol.
+            if (declaredSymbol is IMethodSymbol method &&
+                method.ContainingType.IsRecord &&
+                method.MethodKind == MethodKind.Constructor)
+            {
+                return ImmutableArray<SyntaxNode>.Empty;
+            }
```

Comment:
The two actions that were fired previously had two different `context.ContainingSymbol`s (the type and the primary ctor). Do we know which one we're ignoring now or is it even deterministic?

Reply:
The skipped callback _should_ be for a syntax node, not a symbol. Maybe add some additional tests showing each related callback occurs exactly once for syntax and symbols in a record declaration?

Reply:
@sharwell I know it's for a syntax node, but it's still tied with a symbol I think? I'm referring to [`SyntaxNodeAnalysisContext.ContainingSymbol`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.syntaxnodeanalysiscontext.containingsymbol?view=roslyn-dotnet-4.0.1#Microsoft_CodeAnalysis_Diagnostics_SyntaxNodeAnalysisContext_ContainingSymbol)

Reply:
I had no idea that property existed 🤣 

That said, I believe the type should be the containing symbol.

Reply:
@sharwell Yeah I never needed to use it. All the property usages in Roslyn (per sourceroslyn.io) are only tests. I don't know why it even existed 😄 

> That said, I believe the type should be the containing symbol.

That's exactly what I meant.